src/lib/Components/FilterModal.tsx

```diff
@@ -50,42 +36,165 @@ export class FilterModal extends React.Component<ModalProps, State> {
-                      </Box>
-                    </Flex>
-                    <Sans weight="medium" size="4">
-                      Filter
-                    </Sans>
-                    <Sans size="4">Clear all</Sans>
-                  </Flex>
-                  {this.props.children}
-                  <Button
-                    onPress={() => {
-                      this.closeModal()
+              <ModalBackgroundView>
+                <Flex onTouchStart={this.props.closeModal} style={{ flexGrow: 1 }} />
+                <ModalInnerView visible={isComponentMounted}>
+                  <NavigatorIOS
+                    navigationBarHidden={true}
+                    initialRoute={{
+                      component: FilterOptions,
+                      title: "",
+                      passProps: { closeModal: this.props.closeModal },
                     }}
-                    block
-                    width={100}
-                    variant="secondaryOutline"
-                  >
-                    Ok
-                  </Button>
+                    style={{ flex: 1 }}
+                  />
+                  <Box p={2}>
+                    <Button onPress={() => this.closeModal()} block width={100} variant="secondaryOutline">
+                      Apply
+                    </Button>
+                  </Box>
                 </ModalInnerView>
-              </>
+              </ModalBackgroundView>
             </TouchableWithoutFeedback>
-          </ModalBackgroundView>
-        </TouchableWithoutFeedback>
-      </RNModal>
+          </RNModal>
+        )}
+      </>
+    )
+  }
+}
+
+interface FilterOptionsState {
+  filterOptions: Array<{ type: string; data: any }>
+}
+
+interface FilterOptionsProps {
+  closeModal: () => void
+  navigator: NavigatorIOS
+}
+export class FilterOptions extends React.Component<FilterOptionsProps, FilterOptionsState> {
+  state = {
+    filterOptions: [],
+  }
+
+  componentDidMount() {
+    const filterOptions = []
+
+    filterOptions.push({
+      type: "Sort by",
+      onTap: this.handleNavigationToSortScreen,
+    })
+
+    this.setState({
+      filterOptions,
+    })
+  }
+
+  handleNavigationToSortScreen = () => {
+    this.props.navigator.push({
+      component: SortOptions,
+    })
+  }
+
+  renderFilterOption = ({ item: { type, onTap } }) => {
+    return (
+      <OptionListItem>
+        <Flex p={2} flexDirection="row" justifyContent="space-between" flexGrow={1}>
+          <Serif size="3">{type}</Serif>
+          <Flex data-test-filter-item="item" flexDirection="row" onTouchStart={() => onTap()}>
+            <Serif color={color("black60")} size="3">
+              Default
+            </Serif>
+            <ArrowRightIcon fill="black30" ml={0.3} mt={0.3} />
+          </Flex>
+        </Flex>
+      </OptionListItem>
+    )
+  }
+
+  render() {
+    const { filterOptions } = this.state
+
+    return (
+      <Flex flexGrow={1} data-test-filter-title="item">
```

Comment:
Yes,  I was trying to avoid creating [aliased styled-components](https://github.com/artsy/emission/blob/b958eeb27d9e01df0b4cdce2609abe841f1356fd/src/lib/Components/FilterModal.tsx#L152) simply for test purposes as there are multiple instances of the same component type and I needed a way to grab a specific instance. But it seems like there are trade-offs either way.